src/tools/rust-analyzer/Cargo.toml

```diff
@@ -33,6 +33,7 @@ inherits = "release"
 debug = 2
 
 [patch.'crates-io']
+nix = { git = "https://github.com/psumbera/nix.git", branch = "mypatch-v28" }
```

Comment:
This is not going to happen, this will break our crates-io releases.

Reply:
Is waiting for new `nix` version the only option here? Last `nix` release is from May 2024..

Reply:
Unfortunately I think that's the case, AFAIK neither r-a nor the root manifest here can use cargo patch functionality.

For the root manifest here, I believe we can't be pointing to non-`rust-lang/*` forks via git dependencies too for supply-chain security reasons, cc @pietroalbini.